app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb

```diff
@@ -83,6 +85,20 @@ def resource_restrictions
           resource_restrictions_hash: @resource_restrictions_hash
         )
       end
+
+      def validate_restriction_configuration
+        return unless @restriction_validator
```

Comment:
what's this? when will this be nil?

Reply:
Other authenticators use this class too. And we don't want to add validation to them .

Reply:
well this is entirely not the way we use command classes in our project. why are we validating the restriction as part of the extraction? why aren't we validating it as part of the validation that comes later?